core/shared/src/main/scala/io/circe/Encoder.scala

```diff
@@ -253,6 +255,27 @@ object Encoder extends TupleEncoders with LowPriorityEncoders {
   implicit final val contravariantEncode: Contravariant[Encoder] = new Contravariant[Encoder] {
     final def contramap[A, B](e: Encoder[A])(f: B => A): Encoder[B] = e.contramap(f)
   }
+
+  implicit val encodeByteBuffer: Encoder[ByteBuffer] = instance { bb =>
+    val bytes = if (bb.hasArray){
```

Comment:
Why we don't care about `position` and `remaining` in case if there is an array behind `ByteBuffer`? I'm not sure I have a good answer though, but maybe should just encode the remaining bytes then? I guess that's what users's would expect to happen.